homeassistant/components/media_player/horizon.py

```diff
@@ -92,9 +92,10 @@ def supported_features(self):
     @util.Throttle(MIN_TIME_BETWEEN_SCANS, MIN_TIME_BETWEEN_FORCED_SCANS)
     def update(self):
         """Update State using the media server running on the Horizon."""
-        if self._client.is_powered_on():
-            self._state = STATE_PLAYING
-        else:
+        try:
+            if self._client.is_powered_on():
```

Comment:
This is changing the logic, now if `is_powered_on` is false, we are not resetting the state to off.

Reply:
ah sorry! for the specific issue this PR resolves it would make no difference but you are totally right. this was not the right way and bad style. I think I found a more elegant solution 👍